src/components/scenes/TransactionListScene.tsx

```diff
@@ -74,185 +67,136 @@ function TransactionListComponent(props: Props) {
+      if (typeof listItems[i] === 'string') out.push(i)
     }
-    return sections
-  }, [finalTransactions, loading, searching, spamThreshold])
+    return out
+  }, [listItems])
 
   // ---------------------------------------------------------------------------
   // Side-Effects
   // ---------------------------------------------------------------------------
 
-  React.useEffect(() => {
-    dispatch(fetchMoreTransactions(wallet, currencyCode, reset))
-    if (reset) {
-      setReset(false)
-    }
-  }, [currencyCode, dispatch, reset, wallet])
-
   // Navigate back if the token is disabled from Archive Wallet action
   React.useEffect(() => {
     if (tokenId != null && !enabledTokenIds.includes(tokenId)) {
       navigation.goBack()
     }
   }, [enabledTokenIds, navigation, tokenId])
 
-  // ---------------------------------------------------------------------------
-  // Handlers
-  // ---------------------------------------------------------------------------
-
-  const handleScrollEnd = useHandler(() => {
-    dispatch(fetchMoreTransactions(wallet, currencyCode, reset))
-    if (reset) setReset(false)
-  })
-
-  const handleOnRefresh = useHandler(() => {
-    if (!searching) {
-      setFilteredTransactions(transactions)
-      setSearching(true)
-    }
-  })
-
-  const handleSearchTextChange = useHandler((searchString: string) => {
-    setLoading(true)
-    wallet
-      .getTransactions({
-        currencyCode,
-        searchString
-      })
-      .then(filteredEdgeTransactions => {
-        setFilteredTransactions(
-          transactions.filter(transaction => {
-            return filteredEdgeTransactions.find(item => item.txid === transaction.txid)
-          })
-        )
-        setLoading(false)
-      })
-      .catch(error => {
-        setLoading(false)
-        console.log(error)
-      })
-  })
-
-  const handleSearchingChange = useHandler((isSearching: boolean) => {
-    if (isSearching) return handleOnRefresh()
-    setSearching(false)
-  })
-
   // ---------------------------------------------------------------------------
   // Renderers
   // ---------------------------------------------------------------------------
 
   const refreshControl = React.useMemo(() => {
-    if (finalTransactions.length === 0) return undefined
-    return <RefreshControl refreshing={false} tintColor={theme.searchListRefreshControlIndicator} onRefresh={handleOnRefresh} />
-  }, [finalTransactions.length, handleOnRefresh, theme])
+    return (
+      <RefreshControl
+        refreshing={false}
+        tintColor={theme.searchListRefreshControlIndicator}
+        // useHandler isn't needed, since we're already in useMemo:
+        onRefresh={() => setSearching(true)}
+      />
+    )
+  }, [theme])
 
   const topArea = React.useMemo(() => {
     return (
       <TransactionListTop
-        isEmpty={isTransactionListUnsupported || transactions.length < 1}
+        isEmpty={listItems.length < 1}
```

Comment:
It will be empty once we fully finish loading on an empty wallet with no transactions, or on a wallet that doesn't support the transaction list to begin with.